internal/olm/operator/internal/registry.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"fmt"
+
+	olmclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// SDKLabels are used to identify certain operator-sdk resources.
+var SDKLabels = map[string]string{
+	"owner": "operator-sdk",
+}
+
+// RegistryResources configures creation/deletion of internal registry-related
+// resources.
+type RegistryResources struct {
+	Client  *olmclient.Client
+	Pkg     registry.PackageManifest
+	Bundles []*registry.Bundle
+}
+
+// FEAT(estroz): allow users to specify labels for registry objects.
+
+// CreateRegistryManifests creates all registry objects required to serve
+// manifests from m.manifests in namespace.
+func (m *RegistryResources) CreateRegistryManifests(ctx context.Context, namespace string) error {
+	pkgName := m.Pkg.PackageName
+	binaryKeyValues, err := createConfigMapBinaryData(m.Pkg, m.Bundles)
+	if err != nil {
+		return fmt.Errorf("error creating registry ConfigMap binary data: %w", err)
+	}
+	cm := newRegistryConfigMap(pkgName, namespace,
+		withBinaryData(binaryKeyValues),
+	)
+	volName := getRegistryVolumeName(pkgName)
+	dep := newRegistryDeployment(pkgName, namespace,
+		withRegistryGRPCContainer(pkgName),
+		withVolumeConfigMap(volName, cm.GetName()),
+		withContainerVolumeMounts(volName, []string{containerManifestsDir}),
+	)
+	service := newRegistryService(pkgName, namespace,
+		withTCPPort("grpc", registryGRPCPort),
+	)
+	if err = m.Client.DoCreate(ctx, cm, dep, service); err != nil {
+		return fmt.Errorf("error creating operator %q registry-server objects: %w", pkgName, err)
+	}
```

Comment:
Wdyt about create one and then check, create other and then check again? 
Where it will be called? Will not it be used in a controller, reconcile? 
Should we not first check if the resource exist and then, if not create? 
Could not it face a scenario where the deployment, for example, was already created?

Reply:
Check out the [error handling logic](https://github.com/operator-framework/operator-sdk/blob/master/internal/olm/client/client.go#L90-L95) of `DoCreate()`. "already exists" errors are logged instead of being returned.

Reply:
Have many errors logs in the code that are not actually errors make harder troubleshooting. 
IHMO: would be better:
// check if the resource exists
// if not
// then, build a new resource
// and then create and log

Reply:
@camilamacedo86 I think the `check if resource exists` first will get you a bunch of *misses*. When you are creating things you are likely to not have them there so the check is 90% of the time going to return does not exist. But if you flip the logic the way @estroz has it, you attempt the create, 90% of the time the create will happen, the other 10% it already exists you didn't really spend too much time. As long as the "already exist" error is handled I'm okay with it. I will say I would normally write it like @camilamacedo86 specified because that's just how I do things. But I'm okay with the approach in this PR.